packages/grid/_modules_/grid/hooks/features/virtualization/useGridVirtualScroller.tsx

```diff
@@ -269,9 +270,11 @@ export const useGridVirtualScroller = (props: UseGridVirtualScrollerProps) => {
 
     const rows: JSX.Element[] = [];
 
+    const isEmptySpace = nextRenderContext.lastRowIndex >= currentPage.rows.length;
+
     for (let i = 0; i < renderedRows.length; i += 1) {
       const { id, model } = renderedRows[i];
-
+      const isLastRow = i === renderedRows.length - 1;
```

Comment:
This will be `true` when on the last row of the **current render context**, which is different from the index relative to all visible rows. Is this what you want? The `index` prop gives the "real" index.

Reply:
Hmm, the `index` might also work. The thing is that I need to know if there are fewer rows than the height of the viewport so I can add the border to the last row.

Reply:
I think what you need to do is something that's already being done for the columns. Instead of `columnsMeta.totalWidth`, use `currentPage.rows.length * rowHeight`. To get the height you can try using `apiRef.current.getRootDimensions`.

https://github.com/mui-org/material-ui-x/blob/6208886dbb143f7365f8387ef9d91a6261a3a9d3/packages/grid/_modules_/grid/hooks/features/virtualization/useGridVirtualScroller.tsx#L299-L303

Reply:
But the current way also works. I don't need to calculate the height, I can know if there is space just by checking the indexes.